File: ZTP_Zadanie3/ZTP_Zadanie3/chain/employees/Coach.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ZTP_Zadanie3.chain.wnioski;

namespace ZTP_Zadanie3.pracownicy
{
    class Coach : Employee
    {
        public Coach(Employee employee)
        {
            this.setReceiver(employee);
        }

        public override void receiveARequest(Request request)
        {
            if (isAllowedToReceive(request, "PersonalTrainingRequest"))
            {
                Console.WriteLine("Coach has received " + request.GetType().Name);
            }
            else
            {
                base.receiveARequest(request);
            }
        }
    }
}

File: ChainJava/src/chain/employees/Coach.java
package chain.employees;

import chain.requests.Request;

/**
 * Created by Karolina_2 on 2015-04-01.
 */
public class Coach extends Employee {

    public Coach(Employee employee){
        this.setReceiver(employee);
    }

    @Override
    public void receiveARequest(Request request){
        if(isAllowedToReceive(request, "PersonalTrainingRequest")){
            System.out.println("Coach has received " + request.getClass().getSimpleName());
        }
        else{
            super.receiveARequest(request);
        }
    }
}
